DTS SPI device, DW and quark_se

tbursztyka opened the pull request.

Commit:
dts/bindings: Add bindings for IEEE 802.15.4 CC2520 driver

This is the first driver. For now, only the SPI options can be set
there. In the future also the GPIO lines and interrupts should be set
through DTS.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
dts/x86: Add the 2 master SPI port on curie SoC

This will enable curie based boards to use spi definitions from DTS.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
boards/x86: Enable usage of SPI definition through dts

Tweaking the SPI dw driver to use proper configuration names also.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
arch/x86: Removing unnecessary SPI definitions for quark_se SoC

These are now provided through DTS.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
drivers/ieee802154: Disable options when DTS is in use for CC2520

It does not take crypto part into account yet.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
boards/quark_se_c1000_devboard: Use DTS to enable support of cc2520

Removing relevant parts in Kconfig.defconfig then.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

Commit:
dts/spi: Add a interrupt masq property for DW descriptor

This masq is optional and specific to some architectures.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented on boards/x86/quark_se_c1000_devboard/quark_se_c1000_devboard.dts:
```diff
@@ -54,4 +54,12 @@
+		 spi-max-frequency = <8000000>;
+		 cs-gpios = <&gpio0 11 0>;
+		 label = "CC2520";
+	};
```
@lpereira I guess that's what you would be interested to, as we talked in device driver API meeting

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -136,5 +136,33 @@
+			interrupts = <2 IRQ_TYPE_LEVEL_HIGH 2>;
+			interrupt-parent = <&intc>;
+			interrupt-masq = <0xb0800454>;
+			fifo-depth = <7>;
```
where is fifo-depth specified?  dts binding?

tbursztyka replied:
hum, left over from previous rebase. have to remove that

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -136,5 +136,33 @@
+			label = "SPI_1";
+			interrupts = <3 IRQ_TYPE_LEVEL_HIGH 2>;
+			interrupt-parent = <&intc>;
+			interrupt-masq = <0xb0800458>;
```
where is interrupt-masq defined or used (what binding specifies this?)

tbursztyka replied:
there is a patch adding it.

galak replied:
So interrupt-masq should be a second element in the reg property.  As this is describing a register region that needs access.  You can use 'reg-names' if you want  to have names for the properties as well.  So something like:

```
reg = <0xb0001400 0x400
            0xb0800458 0x4>;
reg-names = <"base" "irq-masq">;
```

galak reviewed (changes requested):
So some of my comments about fifo-depth & interrupt-masq might be cause of github putting the patches in a weird order.

However, the cs-gpio change out of the controller node for SPI isn't acceptable.  The spi controller binding is specified here:

https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/tree/Bindings/spi/spi-bus.txt

And we should conform to it, in which the CS are specified in the controller node, not in the device nodes.

tbursztyka commented on dts/bindings/spi/spi-device.yaml:
```diff
@@ -29,6 +29,11 @@ properties:
       category: required
       description: Maximum clock frequency of device's SPI interface in Hz
       generation: define
+    cs-gpios:
```
No, see my commit message about removing cs-gpios from spi.yaml.

tbursztyka commented:
Ok, this cs-gpios from linux is quite dumb to follow on Zephyr, as it's not the controller which defines it but the spi device. This makes it really counter intuitive.

tbursztyka commented:
github really orders patch weirdly indeed.

galak commented:
> Ok, this cs-gpios from linux is quite dumb to follow on Zephyr, as it's not the controller which defines it but the spi device. This makes it really counter intuitive.

While its a little counter intuitive, it makes sense from a device / device tree point of view.  Since a SPI device doesn't know anything about the controller its connected to it has no idea how the chip selects are implemented -- this is a function of the controller, and thus should live in the controller node.  Otherwise we'd have to update each binding for every combination of chip selection solutions that might exist for all controllers.

Commit:
samples/net: CC2520 is very unlikely to run on top of FRDM-K64F

echo_client auto-builds with CC2520, but that's very unlikely to happen.
Instead, MCR20A is a better device to run on that board.

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

tbursztyka commented:
@galak I reverted back to linux style, and remove the fifo stuff. Can you re-review?

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -136,5 +136,31 @@
 			status = "disabled";
 		};
+
+		spi0: spi@0xb0001000 {
```
spi@0xb0001000 -> spi@b0001000

galak commented on dts/x86/intel_curie.dtsi:
```diff
@@ -136,5 +136,31 @@
+			status = "disabled";
+		};
+
+		spi1: spi@0xb0001400 {
```
similar comment:

spi@0xb0001400 -> spi@b0001400

galak commented on dts/bindings/spi/snps,designware-spi.yaml:
```diff
@@ -32,4 +32,10 @@ properties:
       description: required interrupts
       generation: define
+
+    interrupt-masq:
```
I dont see this used anywhere in this patch set.  Can we drop it for now, I'm still confused what this is suppose to be used for.

tbursztyka replied:
How come you don't see it? Is github messing up things or?

See "dts/x86: Add the 2 master SPI port on curie SoC"
and "boards/x86: Enable usage of SPI definition through dts"

galak replied:
I see it now, I wasn't tracing the #define's enough.

galak commented on samples/net/echo_client/sample.yaml:
```diff
@@ -16,8 +16,8 @@ tests:
-  test_cc2520:
-    extra_args: CONF_FILE="prj_frdm_k64f_cc2520.conf"
+  test_mcr20a:
+    extra_args: CONF_FILE="prj_frdm_k64f_mcr20a.conf"
```
This seems like we should have any entry for prj_frdm_k64f_cc2520.conf and one for prj_frdm_k64f_mcr20a.conf

galak replied:
If we really are removing support for prj_frdm_k64f_cc2520.conf, we should remove the file as well.

tbursztyka replied:
We could indeed. Btw: how to apply dts in samples?

galak reviewed (changes requested):
Looks pretty good a few comments:

1. lets drop interrupt_masq for now since its not used
2. Fix '0x' in dts node names
3. Seems like the change to sample.yaml should support both mcr20a and cc2520 on k64f.

galak reviewed (changes requested):
Now that I understand a bit more about what interrupt-masq is trying to accomplish, we should do that via the reg property, see inline comment.

Also, from the driver point of view this seems to be required, or at least CONFIG_SPI_DW_PORT_n_INT_MASK is required.  So not sure if the code should be:

#ifdef CONFIG_SPI_DW_PORT_n_INT_MASK
_spi_int_mask(CONFIG_SPI_DW_PORT_n_INT_MASK)
#endif

pfalcon commented:
Added @MaureenHelm and @dleach02 to be in loop on frdm_k64f changes. 

(Removed myself from reviewers as I can't say much on the changes.)

galak commented:
@tbursztyka any plans to address comments?

galak commented:
> @tbursztyka any plans to address comments?

Any plans?

galak reviewed (changes requested):
n/a, updated wrong PR

nashif commented:
this was covered in #10249
